packages/common/src/MultiVersionTestHelpers.js

```diff
@@ -0,0 +1,131 @@
+ * @param {Object} contextObjects object containing nested objects which provide context on the creation of the constructor
+ * params. Note each key type defined in the requiredContextObjects must be provided to correctly use this method.
+ * @param {Object} overrideConstructorParams optional override for the constructor params generated.
+ * @returns {Object} version compatible constructor parameters.
+ */
+async function createConstructorParamsForContractVersion(
+  contractVersion,
+  contextObjects,
+  overrideConstructorParams = {}
+) {
+  assert(
+    contractVersion && contractVersion.contractVersion && contractVersion.contractType,
+    "contractVersion must be provided, containing both a contract version and type"
+  );
+  const requiredContextObjects = [
+    "convertSynthetic",
+    "finder",
+    "collateralToken",
+    "syntheticToken",
+    "identifier",
+    "fundingRateIdentifier",
+    "timer",
+    "store",
+    "configStore"
+  ];
+
+  // Check that each of the expected keys is present and not null.
+  requiredContextObjects.forEach(expectedKey => {
+    assert.isTrue(
+      contextObjects[expectedKey] && Object.keys(contextObjects).includes(expectedKey),
+      `Provided context object is missing type ${expectedKey} or is undefined`
+    );
+  });
+
+  let constructorParams = {
+    expirationTimestamp: (await contextObjects.timer.getCurrentTime()).toNumber() + 100000,
+    withdrawalLiveness: "1000",
+    collateralAddress: contextObjects.collateralToken.address,
+    tokenAddress: contextObjects.syntheticToken.address,
+    finderAddress: contextObjects.finder.address,
+    priceFeedIdentifier: padRight(utf8ToHex(contextObjects.identifier), 64),
+    liquidationLiveness: "1000",
+    collateralRequirement: { rawValue: toWei("1.2") },
+    disputeBondPercentage: { rawValue: toWei("0.1") },
+    sponsorDisputeRewardPercentage: { rawValue: toWei("0.1") },
+    disputerDisputeRewardPercentage: { rawValue: toWei("0.1") },
+    minSponsorTokens: { rawValue: contextObjects.convertSynthetic("5") },
+    timerAddress: contextObjects.timer.address,
+    excessTokenBeneficiary: contextObjects.store.address,
+    financialProductLibraryAddress: ZERO_ADDRESS
+  };
+
+  if (contractVersion.contractVersion == "1.2.2") {
+    constructorParams.disputerDisputeRewardPct = constructorParams.disputerDisputeRewardPercentage;
+    constructorParams.sponsorDisputeRewardPct = constructorParams.sponsorDisputeRewardPercentage;
+    constructorParams.disputeBondPct = constructorParams.disputeBondPercentage;
+  }
+
+  if (contractVersion.contractType == "Perpetual") {
+    constructorParams.fundingRateIdentifier = padRight(utf8ToHex(contextObjects.fundingRateIdentifier), 64);
+    constructorParams.configStoreAddress = contextObjects.configStore.address;
+    constructorParams.tokenScaling = { rawValue: toWei("1") };
+  }
+
+  return { ...constructorParams, ...overrideConstructorParams };
```

Comment:
yup. for now this is the case. we might want to make it more restrictive in the future but for now I think that it's fine seeing it will work with all older contract types that we currently have.